### ISSUE TITLE
[Bug]: During concurrent reading and writing, query failed: segment not found: segment not loaded

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#29128]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20231211-3c32ba24-amd64
- Deployment mode(standalone or cluster): cluster 
- MQ type(rocksmq, pulsar or kafka):   pulsar 
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.3.3.post1.dev8
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. deploy milvus and enable level zero segment
```
  config:
    dataCoord:
      enableActiveStandby: true
      segment:
        enableLevelZero: true
    indexCoord:
      enableActiveStandby: true
    log:
      level: debug
    mixCoord:
      enableActiveStandby: true
    queryCoord:
      enableActiveStandby: true
    rootCoord:
      enableActiveStandby: true
```

2. create collection `fouram_ek7rOCO6` -> index -> insert 10m-128d -> flush -> index -> load
3. concurrent search + scene_insert_delete_flush
scene_insert_delete_flush: insert 200 entities -> delete 100 entities -> flush

4. some search failed:
```
[2023-12-11 12:37:43,149 - ERROR - fouram]: RPC error: [search], <MilvusException: (code=65535, message=fail to search on QueryNode 6: worker(6) query failed: segment not found: segment not loaded[segment=446246919104568081])>, <Time:{'RPC start': '2023-12-11 12:37:42.317805', 'RPC error': '2023-12-11 12:37:43.149221'}> (decorators.py:128)
[2023-12-11 12:37:43,150 - ERROR - fouram]: (api_response) : <MilvusException: (code=65535, message=fail to search on QueryNode 6: worker(6) query failed: segment not found: segment not loaded[segment=446246919104568081])> (api_request.py:54)
``` 

### Expected Behavior

all search succ

### Steps To Reproduce

```markdown
argo workflow: https://argo-workflows.zilliz.cc/archived-workflows/qa/8c053cad-b1a1-470b-b378-91561ea4ab19?nodeId=level-zero-stable-insert-ds
```

### Milvus Log

- 4am pods:
```
level0-insert-2-milvus-datanode-866cb54644-6f82t                  1/1     Running                           0                  14h     10.104.29.55    4am-node35   <none>           <none>
level0-insert-2-milvus-datanode-866cb54644-w7ppt                  1/1     Running                           0                  14h     10.104.26.54    4am-node32   <none>           <none>
level0-insert-2-milvus-indexnode-76b98d4798-cpjzk                 1/1     Running                           0                  14h     10.104.30.64    4am-node38   <none>           <none>
level0-insert-2-milvus-mixcoord-778548f49c-p6vqg                  1/1     Running                           0                  14h     10.104.1.110    4am-node10   <none>           <none>
level0-insert-2-milvus-proxy-59d5bf7659-9vpjr                     1/1     Running                           0                  14h     10.104.31.12    4am-node34   <none>           <none>
level0-insert-2-milvus-querynode-5c66b87544-26pxh                 1/1     Running                           0                  14h     10.104.17.161   4am-node23   <none>           <none>
level0-insert-2-milvus-querynode-5c66b87544-nkrq2                 1/1     Running                           0                  14h     10.104.28.85    4am-node33   <none>           <none>
```

- grafana: https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=prometheus&var-cluster=&var-namespace=qa-milvus&var-instance=level0-insert-2&var-collection=All&var-app_name=milvus&from=1702296000000&to=1702319422000

**Comment by yah01** [comment]
/assign @ThreadDao 
fixed by #29130

**Comment by ThreadDao** [comment]
image `master-20231214-264505b7-amd64` fixed
